linkerd/app/outbound/src/ingress.rs

```diff
@@ -61,185 +71,202 @@ impl Outbound<svc::ArcNewHttp<http::Endpoint>> {
+                                        None => Target::Forward(orig_dst),
+                                        Some(a) => {
+                                            let dst =
+                                                NameAddr::from_authority_with_default_port(&a, 80)
+                                                    .map_err(|_| InvalidOverrideHeader)?;
+                                            Target::Override(dst)
+                                        }
+                                    };
+                                Ok(Http {
+                                    target,
+                                    version: protocol,
+                                })
+                            }
+                        },
+                    ))
+                    .push(http::NewNormalizeUri::layer())
+                    .push_on_service(
+                        svc::layers()
+                            .push(http::MarkAbsoluteForm::layer())
+                            // The concurrency-limit can force the service into
+                            // fail-fast, but it need not be driven to readiness on a
+                            // background task (i.e., by `SpawnReady`).  Otherwise, the
+                            // inner service is always ready (because it's a router).
+                            .push(svc::ConcurrencyLimitLayer::new(*max_in_flight_requests))
+                            .push(svc::FailFast::layer("Ingress server", *dispatch_timeout))
+                            .push(rt.metrics.http_errors.to_layer()),
+                    )
+                    .push(http::ServerRescue::layer(config.emit_headers))
+                    .push_on_service(
+                        svc::layers()
+                            .push(http_tracing::server(rt.span_sink.clone(), trace_labels()))
+                            .push(http::BoxResponse::layer())
+                            .push(http::BoxRequest::layer()),
+                    )
+                    .instrument(|a: &http::Accept| debug_span!("http", v = %a.protocol));
+
+                // HTTP detection is **always** performed. If detection fails, then we
+                // use the `fallback` stack to process the connection by its original
+                // destination address.
+                new_http
+                    .push(http::NewServeHttp::layer(*h2_settings, rt.drain.clone()))
+                    .push_switch(
+                        |(http, t): (Option<http::Version>, T)| -> Result<_, Infallible> {
+                            let accept = tcp::Accept::from(t.param());
+                            if let Some(version) = http {
+                                Ok(svc::Either::A(http::Accept::from((version, accept))))
+                            } else {
+                                Ok(svc::Either::B(accept))
+                            }
```

Comment:
very unimportant code golf: a `match` would actually be one line shorter here:
```suggestion
                            match http {
                                Some(version) => Ok(svc::Either::A(http::Accept::from((version, accept)))),
                                None => Ok(svc::Either::B(accept)),
                            }
```

Reply:
rustfmt makes the match statement longer, though.

I think the shortest form is actually
```rust
                            if let Some(version) = http {
                                return Ok(svc::Either::A(http::Accept::from((version, accept))));
                            }
                            Ok(svc::Either::B(accept))
```